client/src/modules/stock/StockEntryExitType.service.js

```diff
@@ -0,0 +1,100 @@
+angular.module('bhima.services')
+.service('StockEntryExitTypeService', StockEntryExitTypeService);
+
+StockEntryExitTypeService.$inject = [];
+
+function StockEntryExitTypeService() {
+  var $ctrl = this;
+
+  $ctrl.entryExitTypeList = [
+    { label : 'patient',
+      labelKey : 'PATIENT_REG.ENTITY',
+      descriptionKey : 'STOCK.PATIENT_DISTRIBUTION',
+      isEntry : false,
+      allowedKey : 'allow_exit_debtor' },
+
+    { label : 'service',
+      labelKey : 'SERVICE.ENTITY',
+      descriptionKey : 'STOCK.SERVICE_DISTRIBUTION',
+      isEntry : false,
+      allowedKey : 'allow_exit_service' },
+
+    { label : 'depot',
+      labelKey : 'DEPOT.ENTITY',
+      descriptionKey : 'STOCK.DEPOT_DISTRIBUTION',
+      isEntry : false,
+      allowedKey : 'allow_exit_transfer' },
+
+    { label : 'loss',
+      labelKey : 'STOCK.EXIT_LOSS',
+      descriptionKey : 'STOCK.LOSS_DISTRIBUTION',
+      isEntry : false,
+      allowedKey : 'allow_exit_loss' },
+
+    { label : 'purchase',
+      labelKey : 'STOCK.ENTRY_PURCHASE',
+      descriptionKey : 'STOCK_FLUX.FROM_PURCHASE',
+      isEntry : true,
+      allowedKey : 'allow_entry_purchase' },
+
+    { label : 'integration',
+      labelKey : 'STOCK.INTEGRATION',
+      descriptionKey : 'STOCK_FLUX.FROM_INTEGRATION',
+      isEntry : true,
+      allowedKey : 'allow_entry_integration' },
+
+    { label : 'donation',
+      labelKey : 'STOCK.DONATION',
+      descriptionKey : 'STOCK_FLUX.FROM_DONATION',
+      isEntry : true,
+      allowedKey : 'allow_entry_donation' },
+
+    { label : 'transfer_reception',
+      labelKey : 'STOCK.RECEPTION_TRANSFER',
+      descriptionKey : 'STOCK_FLUX.FROM_TRANSFER',
+      isEntry : true,
+      allowedKey : 'allow_entry_transfer' },
+  ];
+
+  $ctrl.getAllowedTypes = getAllowedTypes;
+
+  $ctrl.getEntryTypeList = function getEntryTypeList() {
+    return filterByEntry(true);
+  };
+
+  $ctrl.getExitTypeList = function getExitTypeList() {
+    return filterByEntry(false);
+  };
+
+  /**
+   * @function getAllowedTypes
+   *
+   * @description
+   * return allowed entry/exit feature for a given depot
+   *
+   * @param {object} depot
+   */
+  function getAllowedTypes(depot) {
+    if (!depot || !depot.uuid) { return []; }
```

Comment:
Should this just throw an error instead?

Reply:
I don't think so, we are just handling the case where there is no depot it's doesn't mean that it is an error.